apps/rush-lib/src/api/RushConfiguration.ts

```diff
@@ -228,15 +228,6 @@ export interface IRushConfigurationJson {
   variants?: IRushVariantOptionsJson[];
 }
 
-/**
- * This represents the JSON data structure for the "rush-link.json" data file.
```

Comment:
@D4N14L @iclanton For some trivia, the original motivation of **rush-link.json** was to ensure that `rush link` would produce a correct result even when the person forgot to run `rush update`.  For example, suppose I modify **package.json** to remove a reference to another local project, but I have not run `rush update` yet.  I should be able to run `rush unlink` and `rush link` and get back to a valid installation.  (Although the resulting installation is inconsistent with **package.json**, it is nonetheless a correct and usable state, because Node.js `resolve()` doesn't consider the  **package.json** files.)

The PR seems to remove that feature.  I guess that's okay, given our goal of letting PNPM/Yarn managers manage these symlinks instead of Rush.  But if it's possible to get into a broken state, we should report an error to prevent that.

Reply:
We chatted about this a bit and decided to change the design overall (for Yarn and old PNPM and workspace PNPM):

> The original `rush purge` ran `rush unlink`, so for your current PR it seems sufficient to make it so that:
> 
> 1. `rush purge` also deletes the project `node_modules` folders, as it did before.
> 2. `rush link` succeeds without performing any action, and prints a notice saying that it no longer restores the `node_modules` folders.
> 3. `rush unlink` succeeds without performing any action, and prints out a notice saying you need to use `rush purge` now.